integrations/amazon_bedrock/src/haystack_integrations/components/generators/amazon_bedrock/chat/chat_generator.py

```diff
@@ -529,3 +379,44 @@ def run(
             raise AmazonBedrockInferenceError(msg) from exception
 
         return {"replies": replies}
+
+    @component.output_types(replies=List[ChatMessage])
+    async def run_async(
+        self,
+        messages: List[ChatMessage],
+        streaming_callback: Optional[Callable[[StreamingChunk], None]] = None,
+        generation_kwargs: Optional[Dict[str, Any]] = None,
+        tools: Optional[List[Tool]] = None,
+    ):
+        """
+        Async version of the run method. Completes chats using LLMs hosted on Amazon Bedrock.
+
+        :param messages: List of ChatMessage objects representing the conversation history.
+        :param streaming_callback: Optional callback function for handling streaming responses.
+        :param generation_kwargs: Optional dictionary of generation parameters.
+        :param tools: Optional list of Tool objects that the model can use.
+        :return: Dictionary containing the model's replies as a list of ChatMessage objects.
+        """
+        params, callback = self._prepare_request_params(
+            messages=messages, streaming_callback=streaming_callback, generation_kwargs=generation_kwargs, tools=tools
+        )
+
+        try:
+            async_client_coroutine = await self._get_async_client()
```

Comment:
Hmm this seems to create a new async client each time `run_async` is called. 

We may want to assign the async client to `self.async_client` on `_get_async_client` and then add an initial check. So like:

```python
async def _get_async_client(self):

        if self._async_client is not None:
            return self._async_client

        # ...

        self._async_client = async_session.client( ... )
```

Reply:
or we can initialize the client ini `__init__` like on sync version, your choice ;)

Reply:
solved in on-line session/pair programming